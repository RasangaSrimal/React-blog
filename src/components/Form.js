import React, {useState, useEffect } from 'react';
import APIService from '../APIService';

function Form(props) {

    const [title, setTitle] = useState('')
    const [body, setBody] = useState('')

    useEffect(() => {
        setTitle(props.article.title)
        setBody(props.article.body)
    }, [props.article])
    
    const updateArticle = () => {
        APIService.UpdateArticle(props.article.id, {title, body})
        .then(response => props.updatedInformation(response))
    }

    const addArticle = () => {
        APIService.AddArticle({title, body})
        .then(response => props.addedInformation(response))
    }

    return (
        <div>
            {props.article ? (
                <div data-testid = "form" className = "mb-3">
                    <label htmlFor = "title" className = "form-label">Title</label>
                    <input 
                        type = "text" 
                        className = "form-control" 
                        id = "title" 
                        placeholder = "Please enter the title"
                        value = {title}
                        onChange = {e => setTitle(e.target.value)}
                    />
                    <label htmlFor = "body" className = "form-label pt-3">Description </label>
                    <textarea 
                        className = "form-control" 
                        id = "body" 
                        rows = "5"
                        value = {body}
                        onChange = {e => setBody(e.target.value)}
                    ></textarea>
                    <br/>
                    {
                        props.article.id ? 
                            <button 
                                onClick = {updateArticle} 
                                className = "btn btn-success"
                            >Update Article</button>
                        : <button 
                            onClick = {addArticle} 
                            className = "btn btn-success"
                          >Add Article</button>
                    }
                </div>
            ) : null}
        </div>
    )
}

export default Form;