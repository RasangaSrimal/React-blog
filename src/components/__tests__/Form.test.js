import { render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Form from '../Form';

afterEach(() => {
    cleanup();
});

test('should render update article form', () => {
    const article = {
        body: "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla",
        id: 2,
        title: "qui est esse",
        userId: 1
      }
    render(<Form article={article}/>);
    const FormElement = screen.getByTestId('form')
    expect(FormElement).toBeInTheDocument()
    expect(FormElement).toHaveTextContent('est rerum tempore')
});

test('should render add article form', () => {
    const article = {
        body: "new article",
        title: "new title",
        userId: 1
      }
    render(<Form article={article}/>);
    const FormElement = screen.getByTestId('form')
    expect(FormElement).toBeInTheDocument()
    expect(FormElement).toHaveTextContent('Add Article')
    expect(FormElement).not.toHaveTextContent('Update Article')
});

test('matches snapshot for update article form', () => {
    const article = {
        body: "est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla",
        id: 2,
        title: "qui est esse",
        userId: 1
      };
    const tree = renderer.create(<Form article={article}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('matches snapshot for add article form', () => {
    const article = {
        body: "new article",
        title: "new title",
        userId: 1
      }
    const tree = renderer.create(<Form article={article}/>).toJSON();
    expect(tree).toMatchSnapshot();
});