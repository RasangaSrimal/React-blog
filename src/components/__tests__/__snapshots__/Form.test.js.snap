// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for add article form 1`] = `
<div>
  <div
    className="mb-3"
    data-testid="form"
  >
    <label
      className="form-label"
      htmlFor="title"
    >
      Title
    </label>
    <input
      className="form-control"
      id="title"
      onChange={[Function]}
      placeholder="Please enter the title"
      type="text"
      value="new title"
    />
    <label
      className="form-label pt-3"
      htmlFor="body"
    >
      Description 
    </label>
    <textarea
      className="form-control"
      id="body"
      onChange={[Function]}
      rows="5"
      value="new article"
    />
    <br />
    <button
      className="btn btn-success"
      onClick={[Function]}
    >
      Add Article
    </button>
  </div>
</div>
`;

exports[`matches snapshot for update article form 1`] = `
<div>
  <div
    className="mb-3"
    data-testid="form"
  >
    <label
      className="form-label"
      htmlFor="title"
    >
      Title
    </label>
    <input
      className="form-control"
      id="title"
      onChange={[Function]}
      placeholder="Please enter the title"
      type="text"
      value=""
    />
    <label
      className="form-label pt-3"
      htmlFor="body"
    >
      Description 
    </label>
    <textarea
      className="form-control"
      id="body"
      onChange={[Function]}
      rows="5"
      value=""
    />
    <br />
    <button
      className="btn btn-success"
      onClick={[Function]}
    >
      Update Article
    </button>
  </div>
</div>
`;
